Marlin/src/HAL/AVR/fast_pwm.cpp

```diff
@@ -264,8 +264,13 @@ void set_pwm_duty(const pin_t pin, const uint16_t v, const uint16_t v_size/*=255
         ), COM_CLEAR_SET + invert
       );
 
-      uint16_t top = (timer.n == 2) ? TERN(USE_OCR2A_AS_TOP, *timer.OCRnQ[0], 255) : *timer.ICRn;
-      _SET_OCRnQ(timer.OCRnQ, timer.q, (v * top + v_size / 2) / v_size); // Scale 8/16-bit v to top value
+      uint16_t top;
+      if (timer.n == 2)     // if TIMER2
+        top = TERN(USE_OCR2A_AS_TOP, *timer.OCRnQ[0], 255);
```

Comment:
Although a value is given here to `top`, it is not subsequently used. So this could instead just be…
```cpp
if (timer.n == 2) return;
```
…or the whole block could just be…
```cpp
if (timer.n != 2)
  _SET_OCRnQ(timer.OCRnQ, timer.q, (*timer.ICRn * v + v_size / 2) / v_size);
```

Reply:
Timer 2 is useable with 8bits,  so why would we return if the pin maps to it? 

This should work, but it still needs the 8 bit timer 2 exception. 
if (timer.n != 2) _SET_OCRnQ(timer.OCRnQ, timer.q, (*timer.ICRn * v + v_size / 2) / v_size);

Reply:
This option failed on all timers:
```
if (timer.n != 2) _SET_OCRnQ(timer.OCRnQ, timer.q, (*timer.ICRn * v + v_size / 2) / v_size);
```
The calc needed to be ( value * top / size )  = ( value * *timer.ICRn / v_size ) the issue the added bracing as it was incorrect.   ( value * *timer.ICRn / v_size ) did pass however will still need uint16_t top to address timer2 in fast PWM mode.

Reply:
> why would we return if the pin maps to it?

Well, you understand I was blithely commenting on the code as it was, not the code that was more likely intended.

Reply:
I read way too literally in a given moment, I did after the fact think about the actual state .. duh moments.